docs/conventions.md

```diff
@@ -0,0 +1,45 @@
+# Conventions
+This document plans to illustrate the expectations set upon developers contributing to this project. It is intended for such conventions to be relatively lightweight and not overly restrictive, but to provide a baseline for how the project is structured and maintained.
+
+## Language
+All code is to be written in `TypeScript` unless there is a specific reason to use another language. This is to ensure that the codebase is consistent and that all developers can understand the codebase and sing off the same hymn sheet.
+
+## Commit messages
+This project is following the [Conventional Commits Standard](https://www.conventionalcommits.org/en/v1.0.0/#summary). This is to ensure that commit messages maintain a level of consistency describing:
+
+1. What kind of change it is, e.g. new feature or bugfix
+2. What component they affect
+3. A brief description of the change
+
+- `feat(web): Add new feature to the web component`
+- `fix(parser): Fix bug in the parser component`
+- `chore(repo): Update README.md`
+- `docs(repo): Update conventions.md`
+- `ci(repo): Update CI configuration`
+
+See [Versioning](./versioning.md) on why this is important and why we use it in practice.
+
+## Code standards
+**ALL** code despite language is to be written in the following format:
+
+- 2 **spaces** for indentation
+- No trailing whitespace
+- Must have LF line endings
+- Must have a newline at the end of the file
+- Must comply with the ESLint specifications (typescript)
+
+To aid this, developers can run `pnpm lint` in any component which will run ESLint checks and fix them. You must do this before commiting or the CI will block the PR via a status check.
+
+## Branching naming strategy
+There is none. We are not enforcing any form of branching naming strategy. Branch stragegy itself is Trunk based, more info [in how we do releases](./how-do-we-release.md).
+
+## Test standards
+New code should ideally be covered by a test. Our testing tool of choice is [Vitest](https://vitest.dev/guide/) which is a drop in replacement for Jest tests. It is recommended we use this tool as it is faster and more lightweight than Jest, and the web project is built in Vite, which is super performant.
+
+However, it is appreciated that not all contributors are at the point in their professional careers where they see the point of writing tests or are proficient at writing them. As such, it is not a hard requirement to write tests, but it is highly recommended, and this is the perfect project for you to learn to do this.
```

Comment:
I don't agree with this. If someone has the chops to write code, I think they should write tests too. 

Recommend we rewrite to be: 
```suggestion
Tests are strongly recommended, and if not included, you may be asked to add them. If you don't know how to write tests - treat it as a learning experience, and one of the more senior developers will help if they have availability. 
```

Reply:
I'm in two halves about it. Slipping my more junior shoes on, I would find it mega daunting to be told that in order to make a change you MUST write tests. I would like this to be on a "tests where it makes sense" kind of basis, e.g. the Vue templates for example are extremely hard to test, I'm not going to demand that people must write tests for some buttons.

Functionality such as the factory management and the parser and by some extent the backend yes I agree, tests should be expected there.

Reply:
Doesn't the first sentence leave some flexibility for that - especially the front end tests - which I agree with - those are hard. "**Tests are strongly recommended, and if not included, you may be asked to add them.**"

Reply:
### How we determine what needs tests written
Generally, tests for `web` `.vue` files are not mandatory, however if you have the knowledge on how to test them, please do include them. 

For TypeScript only components (e.g. `backend`, `parsing`) tests are going to be **strongly** encouraged. `web` will be treated on a case by case basis. As of writing, tests in `web` correlate around the factory calculations and various other important parts of the system.

Please do not be dismayed if you are asked to provide tests for your PR. Should a contributor have the time, they will try to help guide you writing them. Take this as a learning experience opportunity!

Reply:
@samsmithnz hows that?

Reply:
Love it. 🚢 it.